Linking fails when built with CLANG

Hi,

Shipwright (current GIT code) now builds with CLANG but is still failing to link:
```
/usr/bin/x86_64-linux-gnu-ld.bfd: soh/CMakeFiles/soh.dir/soh/Enhancements/ExtraTraps.cpp.o: undefined reference to symbol 'floorf@@GLIBC_2.2.5'
/usr/bin/x86_64-linux-gnu-ld.bfd: /usr/lib/x86_64-linux-gnu/libm.so.6: error adding symbols: DSO missing from command line
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```

CLANG version: 
```
manuel@debian:~/src/Shipwright/b4$ clang --version
Debian clang version 21.1.8 (10)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/lib/llvm-21/bin
```

--- a/soh/CMakeLists.txt
+++ b/soh/CMakeLists.txt
@@ -691,7 +691,9 @@ else()
     find_package(Vorbis REQUIRED)
     find_package(Opus REQUIRED)
     find_package(OpusFile REQUIRED)
-     set(ADDITIONAL_LIBRARY_DEPENDENCIES
+    # clang doesn't pull libm in transitively the way gcc does, link it explicitly
+    find_library(MATH_LIBRARY m)
+    set(ADDITIONAL_LIBRARY_DEPENDENCIES
     "libultraship;"
         "torch;"
         SDL2::SDL2
@@ -705,6 +707,9 @@ else()
         ${CMAKE_DL_LIBS}
         Threads::Threads
     )
+    if(MATH_LIBRARY)
+        list(APPEND ADDITIONAL_LIBRARY_DEPENDENCIES ${MATH_LIBRARY})
+    endif()
 endif()
 
 if(USE_ASAN)

